QgisModelBaker/qgismodelbaker.py

```diff
@@ -150,37 +155,51 @@ def unload(self):
         del self.__about_action
 
     def show_generate_dialog(self):
-        self.generate_dlg = GenerateProjectDialog(self.iface, self.ili2db_configuration)
-        self.generate_dlg.setAttribute(Qt.WA_DeleteOnClose)
-        self.generate_dlg.setWindowFlags(self.generate_dlg.windowFlags() | Qt.Tool)
-        self.generate_dlg.show()
-        self.generate_dlg.finished.connect(self.generate_dialog_finished)
-        self.__generate_action.setEnabled(False)
+        if self.generate_dlg:
+            self.generate_dlg.reject()
```

Comment:
when does this happen? and is rejecting the right decision at this time?

Reply:
On clicking on the checked icon (see the screencast). I think yes reject. But I think it does not make a big difference to accept.

Reply:
I understand. I was thinking that we just don't react (or raise it to the foreground) since it's more a "wizard" dialog than a dock one toggles on and off, but this also makes sense.